tracks/ruby/exercises/high-scores/mentoring.md

```diff
@@ -0,0 +1,71 @@
+New concepts: Array query methods (+ chaining methods), instantiating a class, getter method or `attr_reader`.
+
+### Minimal solution for approval
+
+```ruby
+
+# version 2+
+
+class HighScores
+  def initialize(scores)
+    @scores = scores
+  end
+
+  def scores  # no need to change getter method to `attr_reader`
+    @scores
+  end
+
+  def personal_best
+    scores.max
+  end
+
+  def latest
+    scores.last
+  end
+
+  def personal_top
+    scores.sort.reverse.take(3) 
+  end
+
+  # composing a string either with string interpolation or with an if.. else.. statement
+  def report
+    difference = "#{personal_best - latest} short of" unless personal_best == latest
+
+    "Your latest score was #{latest}. That's #{difference} your personal best!".squeeze
+  end
+end
+
+```
+
+### Reasonable approaches
+
+- See the one above
+- Defining `@scores`, `@personal_best` and `personal_top` in the constructor.  
+
+### General notes
+- The Instructions point to a beginner friendly explanation of instantiating a class: 
+http://ruby-for-beginners.rubymonstas.org/writing_classes/initializers.html
+and to the rubydocs Array. 
+- Most submissions use an getter method `scores`, not `attr_reader`. 
```

Comment:
I would make this part of "accepted variations".

Reply:
I have not checked all submissions, and "most" could be an incorrect statement at any point in time.  Would "Many" be more suitable here?  Or "Common"?  Or even "Some" which acknowledges that it is done, without a moving target.

Reply:
I think _some_ would be more appropiate. Removing the quantifier altogether would be best I think:

> There are submissions that use a getter method `scores`, instead of `attr_reader :scores`.

That way it doesn't matter what happens in the future!